1_fetch/src/get_wqp_inventory.R

```diff
@@ -0,0 +1,46 @@
+#' Inventory WQP within area of interest
+#' 
+#' @description Function to inventory WQP sites and observations within the area of interest
+#' 
+#' @param bbox_tbl data frame containing an attribute id and bounding box for each
+#' box sampled within the area of interest. The column bbox is of class list.
+#' @param char_names character string indicating the WQP CharacteristicNames to query
+#' @param site_type character string indicating which water feature type(s) to query
+#' @param year_earliest integer; the earliest year to include in the returned inventory,
+#' defaults to NULL, which returns all available years
+#' @param year_latest integer; the most recent year to include in the returned inventory,
+#' defaults to NULL, which returns all available years
+#' 
+#' @value returns a list of sites from the Water Quality Portal
+#' 
+#' @example inventory_wqp(aoi_poly, "Conductivity", "Stream")
+#' @example inventory_wqp(aoi_poly, "Temperature", "Lake, Reservoir, Impoundment")
+#' 
+
+inventory_wqp <- function(bbox_tbl, char_names, site_type, year_earliest = NULL, year_latest = NULL){
+  
+  # Return bounding box to use for WQP data summary
+  bbox <- bbox_tbl$bbox[[1]]
+  
+  # Inventory available WQP data
+  wqp_inventory <- dataRetrieval::readWQPsummary(
+    bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+    summaryYears = "all",
+    siteType = site_type)
```

Comment:
Here I'm using `readWQPsummary` that retrieves summary data from the WQP. This function has been revamped, I think, within the latest version of {dataRetrieval}. One thing I've noticed, though, is that some of the arguments that a user might want to include when downloading WQP data don't seem to work when retrieving summary data (e.g. `sampleMedia`), so I wonder if I shouldn't revert to using `whatWQPdata` here. 

An example of what I mean - here's `readWQPsummary` when looking for "Sediment" vs "Water" and "Sediment" media types:
```
> readWQPsummary(
+   bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+   characteristicName = c("Specific conductance","Temperature, water"),
+   summaryYears = "all",
+   sampleMedia = "Sediment") %>% 
+   dim()
[1] 14220    19                                                                                                                                       
> readWQPsummary(
+   bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+   characteristicName = c("Specific conductance","Temperature, water"),
+   summaryYears = "all",
+   sampleMedia = c("Water","Sediment")) %>% 
+   dim()
[1] 14220    19                                                                                                                                       
> 
```

versus `whatWQPdata`:

```
> whatWQPdata(
+   bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+   characteristicName = c("Specific conductance","Temperature, water"),
+   summaryYears = "all",
+   sampleMedia = c("Sediment")) %>% 
+   dim()
[1] 14 16
> whatWQPdata(
+   bBox = c(bbox$xmin, bbox$ymin, bbox$xmax, bbox$ymax),
+   characteristicName = c("Specific conductance","Temperature, water"),
+   summaryYears = "all",
+   sampleMedia = c("Water","Sediment")) %>% 
+   dim()
[1] 3364   16
>
```

Reply:
🦇 bat signaling @ldecicco-USGS to comment on the appropriate use of `readWQPsummary()` vs the old `whatWQPsummary()`. Laura- everything relevant should be here in Lauren's comment, no need to try to dig through anything else in this repo or PR.

Reply:
This swagger link shows what can be queried with the summary service:
https://www.waterqualitydata.us/data/swagger-ui/index.html?docExpansion=none&url=/data/v3/api-docs#/Summary%20Monitoring%20Location/summaryMonitoringLocationGetRequest
It definitely has fewer options than many of the other data pulls (sampleMedia, start/end dates, etc aren't in there).

The current "dataProfile" default for dataRetrieval is periodOfRecord (so choose that in the swagger dropdown link). I'm going to add in some argument checks in the next update to dataRetrieval so if a user specifies something like sampleMedia they'll get a warning (or error? not sure yet).

Reply:
Thanks for your input and for sharing the swagger link, @ldecicco-USGS! That's good to know that there are fewer options with the summary service than the other data pulls - I like your idea of adding argument checks that inform the user when a certain argument isn't valid.